diff --git a/dist/index.d.cts b/dist/index.d.cts
index 64907250c0d27e43bc980da3e50f9b564919df42..3f54df246b333472c46b60e68d28c06a1a8ec496 100644
--- a/dist/index.d.cts
+++ b/dist/index.d.cts
@@ -15180,7 +15180,13 @@ interface OptionsTypeScriptParserOptions {
     /**
      * Additional parser options for TypeScript.
      */
-    parserOptions?: Partial<ParserOptions>;
+    parserOptions?: Partial<ParserOptions & {
+        /**
+         * @see https://github.com/typescript-eslint/typescript-eslint/issues/2094#issuecomment-1820936720
+         */
+        warnOnUnsupportedTypeScriptVersion: boolean,
+        EXPERIMENTAL_useProjectService: boolean,
+    }>;
     /**
      * Glob patterns for files that should be type aware.
      * @default ['**\/*.{ts,tsx}']
@@ -15193,6 +15199,10 @@ interface OptionsTypeScriptWithTypes {
      * @see https://typescript-eslint.io/linting/typed-linting/
      */
     tsconfigPath?: string | string[];
+    /**
+     * Don't detect `tsconfigPath` anyway.
+     */
+    notDetectTsconfig?: boolean
 }
 interface OptionsHasTypeScript {
     typescript?: boolean;
diff --git a/dist/index.d.ts b/dist/index.d.ts
index 64907250c0d27e43bc980da3e50f9b564919df42..3f54df246b333472c46b60e68d28c06a1a8ec496 100644
--- a/dist/index.d.ts
+++ b/dist/index.d.ts
@@ -15180,7 +15180,13 @@ interface OptionsTypeScriptParserOptions {
     /**
      * Additional parser options for TypeScript.
      */
-    parserOptions?: Partial<ParserOptions>;
+    parserOptions?: Partial<ParserOptions & {
+        /**
+         * @see https://github.com/typescript-eslint/typescript-eslint/issues/2094#issuecomment-1820936720
+         */
+        warnOnUnsupportedTypeScriptVersion: boolean,
+        EXPERIMENTAL_useProjectService: boolean,
+    }>;
     /**
      * Glob patterns for files that should be type aware.
      * @default ['**\/*.{ts,tsx}']
@@ -15193,6 +15199,10 @@ interface OptionsTypeScriptWithTypes {
      * @see https://typescript-eslint.io/linting/typed-linting/
      */
     tsconfigPath?: string | string[];
+    /**
+     * Don't detect `tsconfigPath` anyway.
+     */
+    notDetectTsconfig?: boolean
 }
 interface OptionsHasTypeScript {
     typescript?: boolean;
