diff --git a/dist/index.d.cts b/dist/index.d.cts
index 054ef1e86a49d76248087894f57b6d47e05af143..9bb95f6b80aca4d4e66516fbbeb1704da6a13d07 100644
--- a/dist/index.d.cts
+++ b/dist/index.d.cts
@@ -211,7 +211,13 @@ interface OptionsTypeScriptParserOptions {
     /**
      * Additional parser options for TypeScript.
      */
-    parserOptions?: Partial<ParserOptions>;
+    parserOptions?: Partial<ParserOptions & {
+        /**
+         * @see https://github.com/typescript-eslint/typescript-eslint/issues/2094#issuecomment-1820936720
+         */
+        warnOnUnsupportedTypeScriptVersion: boolean,
+        EXPERIMENTAL_useProjectService: boolean,
+    }>;
     /**
      * Glob patterns for files that should be type aware.
      * @default ['**\/*.{ts,tsx}']
@@ -224,6 +230,10 @@ interface OptionsTypeScriptWithTypes {
      * @see https://typescript-eslint.io/linting/typed-linting/
      */
     tsconfigPath?: string | string[];
+    /**
+     * Don't detect `tsconfigPath` anyway.
+     */
+    notDetectTsconfig?: boolean
 }
 interface OptionsHasTypeScript {
     typescript?: boolean;
diff --git a/dist/index.d.ts b/dist/index.d.ts
index 054ef1e86a49d76248087894f57b6d47e05af143..9bb95f6b80aca4d4e66516fbbeb1704da6a13d07 100644
--- a/dist/index.d.ts
+++ b/dist/index.d.ts
@@ -211,7 +211,13 @@ interface OptionsTypeScriptParserOptions {
     /**
      * Additional parser options for TypeScript.
      */
-    parserOptions?: Partial<ParserOptions>;
+    parserOptions?: Partial<ParserOptions & {
+        /**
+         * @see https://github.com/typescript-eslint/typescript-eslint/issues/2094#issuecomment-1820936720
+         */
+        warnOnUnsupportedTypeScriptVersion: boolean,
+        EXPERIMENTAL_useProjectService: boolean,
+    }>;
     /**
      * Glob patterns for files that should be type aware.
      * @default ['**\/*.{ts,tsx}']
@@ -224,6 +230,10 @@ interface OptionsTypeScriptWithTypes {
      * @see https://typescript-eslint.io/linting/typed-linting/
      */
     tsconfigPath?: string | string[];
+    /**
+     * Don't detect `tsconfigPath` anyway.
+     */
+    notDetectTsconfig?: boolean
 }
 interface OptionsHasTypeScript {
     typescript?: boolean;
